sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -92,6 +93,25 @@ object IntervalUtils {
   }
 
   private val yearMonthPattern = "^([+|-])?(\\d+)-(\\d+)$".r
+  private val yearMonthStringPattern =
+    "^(INTERVAL\\s+)([+|-])?(')([+|-])?(\\d+)-(\\d+)(')(\\s+YEAR\\s+TO\\s+MONTH)$".r
+
+  def castStringToYMInterval(input: UTF8String): Int = {
+    input.trimAll().toString.toUpperCase(Locale.ROOT) match {
```

Comment:
You can specify that your pattern is case insensitive, and remove `.toUpperCase(Locale.ROOT)`

Reply:
You don't see any tests that check proper spaces handling - I mean `trimAll()`. Could you add such checks.

Reply:
Just wonder why don't you trim whitespaces by regular expression?

Reply:
> regular

Match case ` -1-0 `